x-pack/plugin/security/src/test/java/org/elasticsearch/xpack/security/authc/esnative/ReservedRealmTests.java

```diff
@@ -125,10 +126,22 @@ public void testAuthenticationDisabledUserWithStoredPassword() throws Throwable
         verifySuccessfulAuthentication(false);
     }
 
+    public void testLogDeprecatedUser() throws Exception {
```

Comment:
I don't love this test because it's fixed on the single deprecated `kibana` user, and won't work once we actually remove this user. I'm open to suggestions on how to rewrite this more generically, because it wasn't immediately obvious to me.

I could make the `logDeprercatedUser` method public so that it's testable on its own, but that didn't seem like the right solution either.

Reply:
I think this is fine, but I would go with:
```
--- a/x-pack/plugin/security/src/test/java/org/elasticsearch/xpack/security/authc/esnative/ReservedRealmTests.java
+++ b/x-pack/plugin/security/src/test/java/org/elasticsearch/xpack/security/authc/esnative/ReservedRealmTests.java
@@ -126,16 +126,13 @@ public class ReservedRealmTests extends ESTestCase {
         verifySuccessfulAuthentication(false);
     }

-    public void testLogDeprecatedUser() throws Exception {
-        final User expected = new KibanaUser(true);
-        this.verifySuccessfulAuthenticationForUser(expected, true);
-        assertWarnings("The user [kibana] is deprecated and will be removed in a future version of Elasticsearch. " +
-            "Please use the [kibana_system] user instead.");
-    }
-
     private void verifySuccessfulAuthentication(boolean enabled) throws Exception {
         final User expectedUser = randomReservedUser(enabled);
         this.verifySuccessfulAuthenticationForUser(expectedUser, enabled);
+        if (new KibanaUser(enabled).equals(expectedUser)) {
+            assertWarnings("The user [kibana] is deprecated and will be removed in a future version of Elasticsearch. " +
+                    "Please use the [kibana_system] user instead.");
+        }
     }

     private void verifySuccessfulAuthenticationForUser(User expectedUser, boolean enabled) throws Exception {
```